lib/fencing/st_client.c

```diff
@@ -442,17 +442,22 @@ make_args(const char *action, const char *victim, uint32_t victim_nodeid, GHashT
             alias = g_hash_table_lookup(port_map, victim);
         }
 
-        /* Always supply the node's name too:
-         *    https://fedorahosted.org/cluster/wiki/FenceAgentAPI
-         */
-        append_const_arg("nodename", victim, &arg_list);
```

Comment:
But a 'port' would still be produced and passed to fence_legacy and then to the stonith agent even if it's not explicitly configured , which could cause confusion if it's a parameter supported by that agent. So probably we shouldn't produce and pass the parameters reserved by fence-agents to fence_legacy at all.